src/convenience/webhook.ts

```diff
@@ -59,11 +81,166 @@ export function webhookCallback<C extends Context = Context>(
     timeoutMilliseconds?: WebhookOptions["timeoutMilliseconds"],
     secretToken?: WebhookOptions["secretToken"],
 ): (...args: any[]) => any;
+
 export function webhookCallback<C extends Context = Context>(
     bot: Bot<C>,
     adapter?: SupportedFrameworks | FrameworkAdapter,
     webhookOptions?: WebhookOptions,
 ): (...args: any[]) => any;
+
+export function webhookCallback<
```

Comment:
Do you think that there is a way to implement this without an overload per adapter? Ideally, I would like to have a union type here that gets narrowed down based on the adapter name or something. That way, we don't have to duplicate the list of adapters to several places.

Reply:
I have been trying to do this for a several hours now but I can't get TypeScript to properly narrow down the return type. It can narrow down the parameters fine.

```ts
export function webhookCallback<C extends Context = Context>(
    bot: Bot<C>,
    adapter: keyof typeof nativeAdapters,
    webhookOptions?: WebhookOptions,
): (
    ...args: Parameters<typeof nativeAdapters[typeof adapter]>
) => ReturnType<
    typeof nativeAdapters[typeof adapter]
>["handlerReturn"];
```

Any ideas?